### ISSUE TITLE
Add method to set tile count limit when working with OfflineRegionManager

### ISSUE THREAD

**Issue by RodrigoSAlves** [mapbox/mapbox-maps-ios#1099]
## New Feature
Developers should be able to set the **tile count limit** when using the legacy **OfflineRegionManager** class to download **OfflineRegions**.
There used to be a way to set this limit pior on versions preceding version 10.0.0, and android already has a method for this:
https://docs.mapbox.com/android/maps/api/10.0.0/mapbox-maps-android/com.mapbox.maps/-offline-region-manager(android-jvm)/#com.mapbox.maps/OfflineRegionManager/setOfflineMapboxTileCountLimit/#long/PointingToDeclaration/

As far as my knowledge, currently, there is no method that can suffice this requirement on the iOS SDK.

## Why
This is necessary otherwise we will not be able to adjust how many tiles a user can download.
Currently it is set as 6000, but we might want to decrease or increase this number according to our own application's business logic.

**Comment by tomascern** [comment]
Any update on this?

**Comment by RodrigoSAlves** [comment]
Hi @tyofemi,

Just saw that this is being postponed to version 10.6 which has an expected release date of June.
This is tied to a really important feature of our app. We've been waiting for a couple of months now and our users are getting frustrated because they cannot download the tiles / offline regions that were announced when they first registered in the app.
Any chance that this could be included for 10.5?
We have already gotten around several limitations of the new 10.x tile region / offline imagery API's ... We had hopes that at least the legacy version would work but without this our hands are tied...

**Comment by maios** [comment]
Hi @RodrigoSAlves, this is actually available in MapboxCoreMaps v10.5 named [OfflineRegionManager.setOfflineMapboxTileCountLimitForLimit](https://docs.mapbox.com/ios/maps/api/10.5.0-rc.1/core/Classes/MBMOfflineRegionManager.html#/c:objc(cs)MBMOfflineRegionManager(im)setOfflineMapboxTileCountLimitForLimit:)

Please make sure to `import MapboxCoreMaps` to access the said method.
```
import MapboxCoreMaps

offlineRegionManager.setOfflineMapboxTileCountLimitForLimit(limit)
```
